packages/stylist-core/src/lib.rs

```diff
@@ -13,3 +13,78 @@ pub mod ast;
 #[cfg_attr(documenting, doc(cfg(feature = "parser")))]
 #[cfg(feature = "parser")]
 mod parser;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ast::Sheet;
+    use ast::ToStyleStr;
+
+    #[test]
+    fn test_scoped_complex() {
+        let style: Sheet = r#"
+                background-color: black;
+                .with-class {
+                    color: red;
+                }
+                @media screen and (max-width: 600px) {
+                    color: yellow;
+                }
+                @supports (display: grid) {
+                    display: grid;
+                }
+
+                header, footer {
+                    border: 1px solid black;
+
+                    @supports (max-width: 500px) {
+                        max-width: 500px;
+
+                        @media screen and (max-width: 500px) {
+                            display: flex;
+                            flex-direction: row;
+                        }
+                    }
+                }
+            "#
+        .parse()
+        .expect("Failed to create Style.");
+
+        assert_eq!(
+            style.to_style_str(Some("test-style-cls")),
+            r#".test-style-cls {
+    background-color: black;
+}
+.test-style-cls .with-class {
+    color: red;
+}
+@media screen and (max-width: 600px) {
+    .test-style-cls {
+        color: yellow;
+    }
+}
+@supports (display: grid) {
+    .test-style-cls {
+        display: grid;
+    }
+}
+.test-style-cls header, .test-style-cls footer {
+    border: 1px solid black;
+}
+@supports (max-width: 500px) {
+    .test-style-cls header, .test-style-cls footer {
+        max-width: 500px;
+    }
+}
+@supports (max-width: 500px) {
```

Comment:
I don't think it's possible to deduplicate this `@supports` rule without having to implement a logic that looks at the next adjacent block.

I am fine to overlook this bit without introducing more complicated logic into the `StyleContext`.

Reply:
Fine by me aswell. At the end of the day, the output should resemble the input for easier debugging, but really only needs be correctly understood by the browser.